src/main/java/com/quiz/service/impl/QuizServiceImpl.java

```diff
@@ -66,126 +65,160 @@ public QuizMetaData getQuizMetaData(int quizMetaDataID) {
+        if (quizQuestion != null) {
             return quizQuestion.getQuizMetaData();
-        }else{
+        } else {
             throw new RuntimeException("QuizQuestion not found with ID: " + quizQuestionID);
         }
     }
 
     @Override
-    public String updateQuizMetaData(int quizQuestionID, int quizMetaDataID, QuizMetaData updatedMetaData){
+    public String updateQuizMetaData(int quizQuestionID, int quizMetaDataID, QuizMetaData updatedMetaData) {
         QuizMetaData quizMetaData = quizMetaDataRepository.getQuizMetaDataByID(quizMetaDataID);
         QuizQuestion quizQuestion = quizQuestionRepository.getQuizQuestionByID(quizQuestionID);
-        if(quizMetaData!=null){
-            if(!quizMetaData.isCorrect() && updatedMetaData.isCorrect()){
-                for(QuizMetaData metaData : quizQuestion.getQuizMetaData()){
-                    if(metaData.isCorrect()){
+        if (quizMetaData != null) {
+            if (!quizMetaData.isCorrect() && updatedMetaData.isCorrect()) {
+                for (QuizMetaData metaData : quizQuestion.getQuizMetaData()) {
+                    if (metaData.isCorrect()) {
                         metaData.setCorrect(false);
-                        quizMetaDataRepository.update(metaData,metaData.getId());
+                        quizMetaDataRepository.update(metaData, metaData.getId());
                     }
                 }
-                quizMetaDataRepository.update(updatedMetaData,quizMetaDataID);
+                quizMetaDataRepository.update(updatedMetaData, quizMetaDataID);
                 return "QuizMetaData successfully updated!";
-            }else if(quizMetaData.isCorrect() && !updatedMetaData.isCorrect()){
+            } else if (quizMetaData.isCorrect() && !updatedMetaData.isCorrect()) {
                 throw new RuntimeException("Firstly you must assign the correct answer!");
-            }else{
-                quizMetaDataRepository.update(updatedMetaData,quizMetaDataID);
+            } else {
+                quizMetaDataRepository.update(updatedMetaData, quizMetaDataID);
                 return "QuizMetaData successfully updated!";
             }
-        }else{
+        } else {
             throw new RuntimeException("QuizMetaData not found with ID: " + quizMetaDataID);
         }
     }
 
     @Override
-    public void deleteQuizMetaData(int quizMetaDataID){
+    public void deleteQuizMetaData(int quizMetaDataID) {
         QuizMetaData quizMetaData = quizMetaDataRepository.getQuizMetaDataByID(quizMetaDataID);
-        if(quizMetaData!=null){
+        if (quizMetaData != null) {
             quizMetaDataRepository.delete(quizMetaDataID);
-        }else{
+        } else {
             throw new RuntimeException("QuizMetaData not found with ID: " + quizMetaDataID);
         }
     }
 
+    @Override
+    public String saveQuizMetaData(int quizMetaDataID) {
+        return null;
+    }
 
+    // QuizQuestion operations:
+
+    @Override
```

Comment:
write these methods body and push them after that we will review it one more time.